[GLib] API test `/webkit/WebKitSettings/webkit-settings` is failing

obyknovenius opened the pull request:
#### 1ba6a9fa393e478b5aafa4a0519287bb0ccb7f8d
<pre>
[GLib] API test `/webkit/WebKitSettings/webkit-settings` is failing
<a href="https://bugs.webkit.org/show_bug.cgi?id=259049">https://bugs.webkit.org/show_bug.cgi?id=259049</a>

Reviewed by Michael Catanzaro.

The test was failing because
`webkit_settings_get_load_icons_ignoring_image_load_setting()` and
`webkit_settings_set_load_icons_ignoring_image_load_setting()` are
deprecated and issue a `g_warning()` which is fatal by default in
Glib tests.

The solution is to temporarily remove `G_LOG_LEVEL_WARNING` from fatals.

Also, unlike some other similar deprecated API functions, these two were
not marked as `WEBKIT_DEPRECATED`.

* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webKitSettingsSetProperty):
(webKitSettingsGetProperty):
(webkit_settings_class_init):
* Source/WebKit/UIProcess/API/glib/WebKitSettings.h.in:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitSettings.cpp:
(testWebKitSettings):
(testWebKitSettingsNewWithSettings):
* Tools/TestWebKitAPI/glib/TestExpectations.json:

Canonical link: <a href="https://commits.webkit.org/265946@main">https://commits.webkit.org/265946@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/30f113469f44bed52033c718ec82be09bf60cc76

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/12225 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/12570 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12873 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13970 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11798 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/12255 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/14987 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/12586 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14471 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/12389 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/13182 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10348 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/14387 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10467 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/11089 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/18206 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11548 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/11250 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/14436 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11750 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9691 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10966 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3040 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/15296 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11606 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (changes requested):
The 2022 API is stable. It's too late to remove anything.

Deprecating it properly is a good idea, of course.

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
```diff
@@ -213,16 +215,21 @@ static void webKitSettingsSetProperty(GObject* object, guint propId, const GValu
 
+#if !ENABLE(2022_GLIB_API)
+    ALLOW_DEPRECATED_DECLARATIONS_BEGIN
+#endif
```
The preprocessor guards make this harder to read. I would avoid adding guards unless they are really needed. In this case, suppressing deprecation warnings unnecessarily is harmless, so I would always suppress them.

obyknovenius replied:
Done.

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
```diff
@@ -678,6 +697,7 @@ static void webkit_settings_class_init(WebKitSettingsClass* klass)
             _("Whether to load site icons ignoring image load setting."),
             FALSE,
             readWriteConstructParamFlags);
```
```suggestion
            static_cast<GParamFlags>(readWriteConstructParamFlags | G_PARAM_DEPRECATED));
```

mcatanzaro replied:
If you're in the mood for a drive-by cleanup, this is missing from `enable-java` too. Might as well fix that.

obyknovenius replied:
Done. Thanks for the suggestion!

mcatanzaro reviewed (approved):
Thanks for fixing so many tests!

obyknovenius committed:
[GLib] API test `/webkit/WebKitSettings/webkit-settings` is failing
https://bugs.webkit.org/show_bug.cgi?id=259049

Reviewed by Michael Catanzaro.

The test was failing because
`webkit_settings_get_load_icons_ignoring_image_load_setting()` and
`webkit_settings_set_load_icons_ignoring_image_load_setting()` are
deprecated and issue a `g_warning()` which is fatal by default in
Glib tests.

The solution is to temporarily remove `G_LOG_LEVEL_WARNING` from fatals.

Also, unlike some other similar deprecated API functions, these two were
not marked as `WEBKIT_DEPRECATED`.

* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webKitSettingsSetProperty):
(webKitSettingsGetProperty):
(webkit_settings_class_init):
* Source/WebKit/UIProcess/API/glib/WebKitSettings.h.in:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitSettings.cpp:
(testWebKitSettings):
(testWebKitSettingsNewWithSettings):
* Tools/TestWebKitAPI/glib/TestExpectations.json:

Canonical link: https://commits.webkit.org/265946@main
